### ISSUE TITLE
Use '_stripe_intent_id' in checkout mutation

### ISSUE THREAD

**Issue by falcon0ne** [wp-graphql/wp-graphql-woocommerce#612]
Help Wanted!

Is there a way to use the checkout mutation with a created stripe payment intent? The way it is implemented looks like this:

- Create Payment-intent and integrate stripe element
- Create the order with the checkout-mutation
- Confirm the Payment with stripe

In order to use the [stripe payment element](https://stripe.com/docs/payments/payment-element) the intent is created before the order is stored.

```
mutation MyMutation {
  checkout($input: { 
    ...,
    isPaid: false,
    paymentMethod: 'stripe',
    metaData: [{ key: '_stripe_intent_id', value: 'pi_...'}] 
  } {
    ...
  }
}
```

When this mutation is used, the order gets created, but the status is set to "fail". Unfortunately i was not able to find anything about that in the logs. Just things like that:

```
====Stripe Version: 6.1.0====
====Start Log====
Error: stdClass Object
(
    [token_id] => 
    [customer] => 
    [source] => 
    [source_object] => 
    [payment_method] => 
)

====End Log====
```

- Wordpress 5.9.1
- WooCommerce 6.2.0
- WooCommerce Stripe Gateway 6.1.0
- WpGraphQL WooCommerce 0.10.7

Any suggestions or documentation somewhere?

**Comment by kidunot89** [comment]
@falcon0ne Have you seen the unit test [here](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/develop/tests/wpunit/CheckoutMutationTest.php#L1149)?

**Comment by falcon0ne** [comment]
@kidunot89 Not really...
It seems that is not really possible to use the checkout-mutation when you are not in a Wordpress-context. Currently I'm trying to move on with the create-order mutation and not using a "_stripe_source_id". 

If you use the Stripe-Payment-Element, it is not possible "client-side" to generate a source for a specific customer. Even generating a source is not part of the Payment Element. That means, that the checkout-mutation did not work for that case.

**Comment by scottyzen** [comment]
@falcon0ne I don't know if this will help but this is working for me. I'm using [vue-stripe-elements-plus](https://github.com/ectoflow/vue-stripe-elements) to generate the sourceId.

```javascript
const variables = {
	billing: this.billing,
	shipping: this.shipping ,
	metaData:  [{ key: '_stripe_source_id', value: sourceId }],
	paymentMethod: this.paymentMethod,
	customerNote: this.orderNote,
	shipToDifferentAddress: this.shipToDifferentAddress,
};
const { checkout } = await this.$graphql.default.request(CHECKOUT, variables);
```

```graphql
mutation Checkout($billing: CustomerAddressInput = {}, $metaData: [MetaDataInput] = {key: "", value: ""}, $paymentMethod: String = "stripe", $shipping: CustomerAddressInput = {}, $customerNote: String = "", $shipToDifferentAddress: Boolean = false) {
  checkout(
    input: {paymentMethod: $paymentMethod, billing: $billing, metaData: $metaData, shipping: $shipping, customerNote: $customerNote, shipToDifferentAddress: $shipToDifferentAddress}
  ) {
    result
    redirect
  }
}
```

**Comment by falcon0ne** [comment]
@scottyzen thanks for your reply. It works also for me, when I use the Card Element. But I need to use the [Full Payment Element](https://stripe.com/docs/payments/payment-element). And this one needs a "client_secret" to start and can not generate a soureId. At least I did not found a way to do so.

**Comment by talon2295** [comment]
@falcon0ne I know it's been a long time, but were you able to find a solution? i have the same problem.

**Comment by falcon0ne** [comment]
Hey @talon2295 I don't have a solution.
I switched to the Card-Element and did not use the Full-Payment-Element.

**Comment by talon2295** [comment]
@falcon0ne thanks for your reply.
The problem is that Card-Element doesn't allow for handling payments with 3D Secure authentication. 
I solved it in that way #754

**Comment by falcon0ne** [comment]
@talon2295 i solved it a bit different.
I give back the payment-intent to the frontend in case the payment needs handling via stripe.
The StripeJS in the frontend can then handle the authentication.
